### PR TITLE
Fixed hang

### PR DESCRIPTION
**Pull request by Quaker762**
Fixed Tetris hang.

I'm going to do a complete rework of some of the flags. The flags are the source of all problems...

I hate flags...

### CODE REVIEW

**Comment by wornwinter** [comment]
What happens now when you run it? Does it progress further than the title screen?

**Comment by Quaker762** [comment]
w00t

It gets past the credits but crashes with an uninterrupted exception. The problem was dodgy VBLANK code or weird interrupt handling. The problem now is the register values being all fucked up.

File: src/DMGCPU.cpp [hunk 1]
```diff
@@ -1847,18 +1851,28 @@ void c_DMGCPU::OPCode0x7C()
...
-    UNSET_FLAG_BIT(SUB_BIT);
+    uint8_t result = Registers.AF.hi + Registers.HL.lo;
 
-    if((Registers.AF.hi + Registers.HL.lo) == 0) SET_FLAG_BIT(ZERO_BIT);
-    else UNSET_FLAG_BIT(ZERO_BIT);
-    if(((Registers.AF.hi&0xF) + (Registers.HL.lo&0xF)) > 0xF) SET_FLAG_BIT(HC_BIT);
-    else UNSET_FLAG_BIT(HC_BIT);
-    if((Registers.AF.hi + Registers.HL.lo) > 0xFF) SET_FLAG_BIT(CARRY_BIT);
-    else UNSET_FLAG_BIT(CARRY_BIT);
+    if(result & 0xFF00)
```
Comment (wornwinter) [inline src/DMGCPU.cpp]: Not sure about this. I think that ```if((Registers.AF.hi + Registers.HL.lo) > 0xFF)``` might be a better option, as we need to check if there's a carry from Bit 7.

**Comment by wornwinter** [comment]
I'm nearly sure I made a mistake in BIT 2, b also if you want to look at
that in this pr lol

File: src/DMGCPU.cpp (hunk 1 above)
Comment (Quaker762) [inline src/DMGCPU.cpp]: Yeah I fixed that up. I'm working on making all the opcodes more consistent, so any old stuff like this/2AM programming will be removed.

**Comment by Quaker762** [comment]
@wornwinter Okay I'll get on it. I'm going to catch some Z's (I have a Physics exam on Wednesday), so I'll take a look on Thursday unless you want to fix it up :)

**Comment by wornwinter** [comment]
I'll get to it tonight lol. Just in work at the minute and can't really
check in depth.

**Comment by Quaker762** [comment]
Okay cool. Don't merge by the way I just broke something and it won't load past the Nintendo logo anymore....

**Comment by Quaker762** [comment]
Okay just did a quick check. The problem is at ADD A, (HL). I fucked a flag up. I'll create a PR ASAP

**Comment by Quaker762** [comment]
wut
Okay it gets to the menu, but this is the problem I was talking about. 
It sits there for about 10 seconds (keep in mind there's NO cursor/selection arrow displayed) before it does this..

I honestly have no idea what in the hell would cause this... Any ideas??

**Comment by Quaker762** [comment]
hahaha

BAHAHAHAHAHA

**Comment by wornwinter** [comment]
Haha brilliant. How's it going now?

**Comment by Quaker762** [comment]
Pretty good, I added a few more opcodes, however there's still some strange behaviour occuring (it's not crashing to the game screen which is good). Looks like it's getting stuck in a 'cinoop' style loop, so it's possibly something to do with VBLANK.

**Comment by wornwinter** [comment]
That's a possibility. The issue with the cursor is likely the lack of
sprites. I don't think I implemented that yet.

**Comment by Quaker762** [comment]
Yeah from the looks of it, its trying to write some OAM stuff (I added some
OAM FMA) to unknown IO registers which could be causing the loop if it's
waiting for return value or something.

**Pull request merged**